SKYKitChat/Classes/UI/ConversationView/SKYChatConversationViewController.swift

```diff
@@ -0,0 +1,612 @@
+                                                   readyToSendMessage message: SKYMessage)
+
+    @objc optional func conversationViewController(_ controller: SKYChatConversationViewController,
+                                                   finishSendingMessage message: SKYMessage)
+
+    @objc optional func conversationViewController(_ controller: SKYChatConversationViewController,
+                                                   failedToSendMessageText text: String,
+                                                   date: Date,
+                                                   error: Error)
+}
+
+open class SKYChatConversationViewController: JSQMessagesViewController {
+
+    weak public var delegate: SKYChatConversationViewControllerDelegate?
+
+    public var skygear: SKYContainer = SKYContainer.default()
+    public var conversation: SKYConversation?
+    public var userConversation: SKYUserConversation?
+    public var participants: [String: SKYRecord] = [:]
+    public var messages: [SKYMessage] = []
+    public var messagesFetchLimit: UInt = 25
+    public var typingIndicatorShowDuration: TimeInterval = TimeInterval(5)
+
+    public var messageChangeHandler: ((SKYChatRecordChangeEvent, SKYMessage) -> Void)?
+    public var typingIndicatorChangeHandler: ((SKYChatTypingIndicator) -> Void)?
+    public var typingIndicatorPromptTimer: Timer?
+
+    public var bubbleFactory: JSQMessagesBubbleImageFactory? = JSQMessagesBubbleImageFactory()
+    public var incomingMessageBubble: JSQMessagesBubbleImage?
+    public var outgoingMessageBubble: JSQMessagesBubbleImage?
+
+    public var incomingMessageBubbleColor: UIColor? {
+        didSet {
+            self.incomingMessageBubble = self.bubbleFactory?
+                .incomingMessagesBubbleImage(with: self.incomingMessageBubbleColor)
+        }
+    }
+
+    public var outgoingMessageBubbleColor: UIColor? {
+        didSet {
+            self.outgoingMessageBubble = self.bubbleFactory?
+                .outgoingMessagesBubbleImage(with: self.outgoingMessageBubbleColor)
+        }
+    }
+
+    static let errorDomain: String = "SKYChatConversationViewControllerErrorDomain"
+    var errorCreator: SKYErrorCreator {
+        return SKYErrorCreator(defaultErrorDomain: SKYChatConversationViewController.errorDomain)
+    }
+}
+
+// MARK: - Initializing
+
+extension SKYChatConversationViewController {
+
+    public class func create() -> SKYChatConversationViewController {
+        return SKYChatConversationViewController(nibName: "JSQMessagesViewController",
+                                                 bundle: Bundle(for: JSQMessagesViewController.self))
+    }
+}
+
+// MARK: - Lifecycle
+
+extension SKYChatConversationViewController {
+
+    override open func viewDidLoad() {
+        super.viewDidLoad()
+
+        self.senderId = self.skygear.currentUserRecordID
+
+        // update the display name after fetching participants
+        self.senderDisplayName = "me"
```

Comment:
I am not sure about the convention of this project, but I always wrap user-facing strings with `NSLocalizedString` even we don't have translation